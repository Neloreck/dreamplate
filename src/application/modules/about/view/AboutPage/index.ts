/**
 * @module application/about
 */

export * from "./AboutPage.component";
