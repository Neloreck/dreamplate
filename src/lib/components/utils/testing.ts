/**
 * @module @lib/components
 */

import { CSSResult } from "lit-element";

export const nestedShadowOf = (component: { styles: CSSResult }, ...nested: Array<string>) => {

  let commented: string = "";

  nested.forEach((it: string) => commented += `<!----><!---->${it}<!----><!---->`);

  return `${commented}<style>${component.styles}</style>`;
};
