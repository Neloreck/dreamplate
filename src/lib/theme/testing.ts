import { mount, ShallowWrapper } from "enzyme";
import { createElement, ReactElement } from "react";
import { ThemeProvider } from "react-jss";

// Lib.
import { createDefaultTheme } from "./utils";

/**
 * Utility for rendering react component inside JSS context.
 */
export const withMockedTheme = (element: ReactElement) => createElement(ThemeProvider, { children: element, theme: createDefaultTheme });

/**
 * Utility to get nested in @Styled element.
 */
export const getThemedComponent = (element: ShallowWrapper<any>) => mount(element.getElement()).children().children();
