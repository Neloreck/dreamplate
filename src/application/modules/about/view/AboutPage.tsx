import { useManager } from "dreamstate";
import { ReactElement } from "react";
import { createUseStyles } from "react-jss";
import { Link } from "react-router-dom";

import { RouterManager } from "@/core/data/store";
import { MainHeader } from "@/core/view/components";
import { IApplicationTheme } from "@/lib/theme";

export const useStyles = createUseStyles(({ spacing }: IApplicationTheme) => ({
  content: {
    "& > button": {
      margin: spacing.unit * 2
    },
    alignItems: "center",
    alignSelf: "normal",
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    justifyContent: "center",
    overflowY: "auto",
    padding: spacing.unit * 2
  }
}));

export function AboutPage({
  classes: { content } = useStyles(),
  routerContext: { history } = useManager(RouterManager)
}): ReactElement {
  return (
    <>
      <MainHeader />

      <main className={content}>
        About page.
        <Link to={history.createHref("/home")} reloadDocument>
          <button>Go home</button>
        </Link>
      </main>
    </>
  );
}
