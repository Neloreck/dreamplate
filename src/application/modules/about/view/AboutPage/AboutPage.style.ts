import { createUseStyles } from "react-jss";

// Lib.
import { IApplicationTheme } from "@Lib/theme";

export const useStyles = createUseStyles(({ palette }: IApplicationTheme) => ({
  content: {
    alignItems: "center",
    alignSelf: "normal",
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    justifyContent: "center",
    overflowY: "auto"
  }
}));
