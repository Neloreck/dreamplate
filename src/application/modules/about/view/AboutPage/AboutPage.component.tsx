import { useManager } from "dreamstate";
import { ReactElement, useCallback } from "react";

import { RouterContextManager } from "@Core/data/store";
import { MainHeader } from "@Core/view/components/MainHeader";

import { useStyles } from "./AboutPage.style";

import "@Lib/components/custom/CustomButton";

export function AboutPage({
  classes: { content } = useStyles(),
  routerContext: { routingActions: { hardPush } } = useManager(RouterContextManager)
}): ReactElement {
  const onHomeNavigated = useCallback(() => hardPush("/home"), []);

  return (
    <>

      <MainHeader/>

      <main className={content}>

        About page.

        <custom-button onClick={onHomeNavigated}>
          Go home
        </custom-button>

      </main>

    </>
  );
}
