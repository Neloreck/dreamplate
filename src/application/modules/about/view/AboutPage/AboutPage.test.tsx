import { shallow } from "enzyme";
import { Classes } from "jss";

import { createMockClasses, createMockContext } from "@Lib/testing";

import { AboutPage } from "./AboutPage.component";

describe("About page component rendering.", () => {
  const classes: Classes = createMockClasses();

  it("Should match HOC snapshot.", () => {
    const wrapper = shallow(<AboutPage classes={classes} routerContext={createMockContext()}/>);

    expect(wrapper).toMatchSnapshot("Default HOC layout.");
  });
});
