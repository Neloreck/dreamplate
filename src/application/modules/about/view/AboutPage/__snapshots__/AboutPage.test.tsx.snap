// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page component rendering. Should match HOC snapshot.: Default HOC layout. 1`] = `
<Fragment>
  <MainHeader />
  <main
    className="content"
  >
    About page.
    <button
      onClick={[Function]}
    >
      Go home
    </button>
  </main>
</Fragment>
`;
